resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/LoggingPodStatusWatcher.scala

```diff
@@ -116,17 +120,59 @@ private[kubernetes] class LoggingPodStatusWatcher(podCompletedFuture: CountDownL
       ("start time", pod.getStatus.getStartTime),
       ("container images",
         pod.getStatus.getContainerStatuses()
-            .asScala
-            .map(_.getImage)
-            .mkString(", ")),
+          .asScala
+          .map(_.getImage)
+          .mkString(", ")),
       ("phase", pod.getStatus.getPhase()),
       ("status", pod.getStatus.getContainerStatuses().toString)
     )
+    formatPairsBundle(details)
+  }
 
+  private def formatPairsBundle(pairs: Seq[(String, String)]) = {
     // Use more loggable format if value is null or empty
-    details.map { case (k, v) =>
-      val newValue = Option(v).filter(_.nonEmpty).getOrElse("N/A")
-      s"\n\t $k: $newValue"
+    pairs.map {
+      case (k, v) => s"\n\t $k: ${Option(v).filter(_.nonEmpty).getOrElse("N/A")}"
     }.mkString("")
   }
+
+  override def awaitCompletion(): Unit = {
+    podCompletedFuture.countDown()
```

Comment:
Shouldn't this be `podCompletedFuture.await`? My client exits right away without waiting for the driver pod to complete. I wonder if it's because of this line?

>2017-06-02 14:48:36 INFO  Client:54 - Waiting for application spark-hdfstest-1496440110123 to finish...
2017-06-02 14:48:36 INFO  LoggingPodStatusWatcherImpl:54 - Container final statuses:
	 Container name: spark-kubernetes-driver
	 Container image: docker:5000/spark-driver:kimoon-0602-1
	 Container state: Waiting
	 Pending reason: PodInitializing
2017-06-02 14:48:36 INFO  Client:54 - Application spark-hdfstest-1496440110123 finished.